### PR TITLE
Fix regression in AwsJobStoreCacheTest (resolves #896)

### LINKED ISSUE #896: Regression in AwsJobStoreCacheTest
http://jenkins.cgcloud.info/job/toil-pull-requests/1346/testReport/junit/src.toil.test.src.jobCacheTest/AwsJobStoreCacheTest/testRemoveLocalImmutablyReadFile/

### PR DESCRIPTION
**Pull request by arkal**
resolves #896

Previously failed to account for failure of cleanup jobs in the test.

### CODE REVIEW

**Comment by arkal** [comment]
@hannes-ucsc this fixed the regression. It had 1 random failure in an Azure test and I think that's not cache related (`RuntimeError: Stats/logging collator failed with exit code 1.`).

**Comment by arkal** [comment]
so i'm removing the in progress flag

**Comment by arkal** [comment]
Jenkins, test this please

**Comment by arkal** [comment]
@hannes-ucsc this is ready

File: src/toil/test/src/jobCacheTest.py [hunk 1]
```diff
@@ -702,16 +702,16 @@ def testRemoveLocalImmutablyReadFile(self):
...
             A.addChild(B)
             try:
                 Job.Runner.startToil(A, self.options)
             except FailedJobsException as err:
-                self.assertEqual(err.numberOfFailedJobs, 1)
+                self.assertEqual(err.numberOfFailedJobs, 2)
                 errMsg = self._parseAssertionError(self.options.logFile)
-                if not 'explicitly' in errMsg:
+                if 'explicitly' not in errMsg:
                     self.fail('Shouldn\'t see this')
```
Comment (hannes-ucsc) [inline src/toil/test/src/jobCacheTest.py]: No backslashes, please.

'explicitly' is too unspecific and it doesn't help the reader understand what exactly you are expecting for. And "Shouldn't see this" is pretty much useless as far as an assertion error is concerned.

File: src/toil/test/src/jobCacheTest.py [hunk 2]
```diff
@@ -739,7 +739,7 @@ def _removeReadFileFn(job, fileToDelete, readAsMutable):
                 try:
                     job.fileStore.deleteLocalFile(fileToDelete)
                 except CacheError as err:
-                    if not err.message.contains('explicitly'):
+                    if 'explicitly' not in err.message:
```
Comment (hannes-ucsc) [inline src/toil/test/src/jobCacheTest.py]: Same here. Too unspecific.

**Comment by hannes-ucsc** [comment]
To be continued: https://github.com/BD2KGenomics/toil/issues/911

**Pull request merged**